integration-tests/lib/assertions.py

```
@@ -179,3 +183,41 @@ def assert_podman_secret_name(session: ContainerSession) -> None:
+def assert_directory_contract(session: ContainerSession) -> None:
+    """Assert the provisioned `/var/lib/repovec` data tree contract.
+
+    Every data directory must be `repovec:repovec` with mode `0700` (SI-1),
+    and the `qdrant-storage` child must be `root:root` with mode `0700`
+    (SI-2). Ownership is asserted by name, proving the
+    sysusers-before-tmpfiles ordering resolved the `repovec` account first
+    (R-2).
+    """
+
+    expected = {
+        **{path: (DATA_DIR_MODE, REPOVEC_USER, REPOVEC_GROUP) for path in DATA_DIRS},
```

Ruff (0.16.3)

[warning] Unnecessary dict comprehension for iterable; use `dict.fromkeys` instead

Replace with `dict.fromkeys(iterable)`)

(C420)